src/WebJobs.Script/Description/CSharp/LocalSharedAssemblyProvider.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    /// <summary>
+    /// An <see cref="ISharedAssemblyProvider"/> that resolves assembly name references against local/private references.
+    /// </summary>
+    public class LocalSharedAssemblyProvider : ISharedAssemblyProvider
+    {
+        private readonly string _assembliesPath;
+        private readonly Regex _nameRegex;
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="LocalSharedAssemblyProvider"/>.
+        /// </summary>
+        /// <param name="assemblyNamePattern">The assembly name pattern to validate against. 
+        /// Only names matching the pattern will be resolved by this provider</param>
+        public LocalSharedAssemblyProvider(string assemblyNamePattern)
+        {
+            _nameRegex = new Regex(assemblyNamePattern, RegexOptions.Compiled | RegexOptions.IgnoreCase);
+            _assembliesPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "bin");
+        }
+        public bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            assembly = null;
+
+            if (_nameRegex.IsMatch(assemblyName))
+            {
+                // Check if the assembly is currently loaded:
+                assembly = AppDomain.CurrentDomain
+                    .GetAssemblies()
+                    .FirstOrDefault(a => string.Compare(a.GetName().Name, assemblyName, StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
Only using `LoadFrom` here because this is for shared assemblies anyway (and the additional benefits of bringing in dependencies, etc.), we can switch this to `LoadFile` if we want to load within that context. Just thought that would make sense based on what this intends to expose.